vim-patch:8.1.{579,654,655},9.1.0546

janlazo opened the pull request:
Problem:  vim-tiny fails on CTRL-X/CTRL-A
          (Rob Foehl, after 9.1.0172)
Solution: Move #ifdefs, so that after changing the line in del_bytes,
          the cached textlen value is invalidated

closes: vim/vim#15178

https://github.com/vim/vim/commit/03acd4761be1c2766d3ec17534ea63cdf8dd565d

<!--
  Thank you for contributing to Neovim!
  If this is your first time, check out https://github.com/neovim/neovim/blob/master/CONTRIBUTING.md#pull-requests-prs
  for our PR guidelines.
-->

----

Trying to track down problems with my port v8.2.0845 based on the N/A patches that affect memline.c

- 8.1.0581 is still N/A is because of 9.0.0013 (https://github.com/neovim/neovim/pull/12662)

janlazo committed:
vim-patch:9.1.0546: vim-tiny fails on CTRL-X/CTRL-A

Problem:  vim-tiny fails on CTRL-X/CTRL-A
          (Rob Foehl, after 9.1.0172)
Solution: Move #ifdefs, so that after changing the line in del_bytes,
          the cached textlen value is invalidated

closes: vim/vim#15178

https://github.com/vim/vim/commit/03acd4761be1c2766d3ec17534ea63cdf8dd565d

Co-authored-by: Christian Brabandt <cb@256bit.org>
Co-authored-by: zeertzjq <zeertzjq@outlook.com>

janlazo committed:
vim-patch:8.1.0579: cannot attach properties to text

Problem:    Cannot attach properties to text.
Solution:   First part of adding text properties.

https://github.com/vim/vim/commit/98aefe7c3250bb5d4153b994f878594d1745424e

"textprop" feature remains N/A.
Porting for ml_replace_len().

Co-authored-by: Bram Moolenaar <Bram@vim.org>

zeertzjq commented on test/functional/legacy/028_source_ctrl_v_spec.lua:
```diff
@@ -35,6 +35,21 @@ describe('CTRL-V at the end of the line', function()
+
+      12354]])
+    feed_command('/12352')
+    feed('')
+    feed_command('/12354')
+    feed('')
```
`feed('')` doesn't do anything:
```lua
local function nvim_feed(input)
  while #input > 0 do
```

janlazo replied:
Could try <C-A>, <C-X> instead but I'm using `feed()`.

janlazo committed:
vim-patch:8.1.0654: when deleting a line text property flags are not adjusted

Problem:    When deleting a line text property flags are not adjusted.
Solution:   Adjust text property flags in preceding and following lines.

https://github.com/vim/vim/commit/c1a9bc1a7284bd0e60f9bddfef6a4ee733bfc838

"textprop" feature remains N/A.
Porting to sync ml_delete_int() with Vim 8.2.0845 and later.

Co-authored-by: Bram Moolenaar <Bram@vim.org>

janlazo committed:
vim-patch:8.1.0655: when appending a line text property flags are not added

Problem:    When appending a line text property flags are not added.
Solution:   Add text properties to a newly added line.

https://github.com/vim/vim/commit/b56ac049ea6ca77a0a9b0a415bac5e82ae60b842

Co-authored-by: Bram Moolenaar <Bram@vim.org>
